### ISSUE TITLE
If the pnpm install gets corrupted, it should be an error not a warning

### ISSUE THREAD

**Issue by pgonzal** [pnpm/pnpm#1132]
### Code to reproduce the issue:

On our CI machines we've been seeing occasional errors like this:

```
2018-04-30T22:22:10.5694107Z Resolving: total 2303, reused 2171, downloaded 131
2018-04-30T22:22:11.5688415Z Resolving: total 2303, reused 2171, downloaded 131
2018-04-30T22:22:12.5694198Z Resolving: total 2303, reused 2171, downloaded 132
2018-04-30T22:22:13.1194664Z WARN  There's a directory in node_modules without package.json: C:\agent\2\_work\14\s\common\temp\node_modules\.onedrive.pkgs.visualstudio.com\react-slick\0.23.1\react-dom@15.6.2+react@15.6.2\node_modules\react-dom
2018-04-30T22:22:13.3256519Z WARN  There's a directory in node_modules without package.json: C:\agent\2\_work\14\s\common\temp\node_modules\.onedrive.pkgs.visualstudio.com\react-addons-test-utils\15.4.2\react-dom@15.6.2\node_modules\react-dom
```

This error happens even after deleting the `node_modules` folder, but seems to go away if we delete the PNPM store folder.  I suspect it may occur when the installation is forcibly terminated, e.g. because a PR build was canceled because an update was pushed to the Git branch.  (So far I haven't proven that this is actually PNPM's fault; if I do I will open a separate GitHub issue.)

### Expected behavior:

When this corruption is detected, the PNPM install should fail with a nonzero exit code.

### Actual behavior:

PNPM prints a warning and appears to succeed.  The resulting node_modules folder has broken symlinks.

### Additional information:

 - `node -v` prints: 8.9.4
 - Windows, OS X, or Linux?: Windows

**Comment by pgonzal** [comment]
If the [--warnings-as-errors feature request #1126](https://github.com/pnpm/pnpm/issues/1126) gets implemented, that would solve this problem as well.

**Comment by pgonzal** [comment]
I found a deterministic repro for the `There's a directory in node_modules without package.json` error.  I've opened  [issue 1133](https://github.com/pnpm/pnpm/issues/1133).

**Comment by pgonzal** [comment]
CC @nickpape-msft @iclanton

**Comment by zkochan** [comment]
Closing this as the initial issue was fixed and #1126 was created to make strict warnings
